### PR TITLE
WIP: Update mongodb indexes based on utilized queries

### PR DESCRIPTION
**Pull request by hiliang-cmu**
TODO Profiling:

- problems
- submissions

### CODE REVIEW

File: picoCTF-web/api/db.py [hunk 1]
```diff
@@ -87,13 +100,15 @@ def index_mongo():
...
-    db.teams.create_index("size",
+    db.teams.create_index(
+        "size",
         name="non-empty size",
         partialFilterExpression={
             "size": {
                 "$gt": 0
             }
         })
+    db.teams.create_index("country")
```
Comment (hiliang-cmu) [inline picoCTF-web/api/db.py]: :O

Comment (dmartin) [reply picoCTF-web/api/db.py]: lol

File: picoCTF-web/api/db.py [hunk 2]
```diff
@@ -53,6 +55,12 @@ def index_mongo():
     log.debug("Ensuring mongo is indexed.")
 
     db.exceptions.create_index([('time', pymongo.DESCENDING)])
+    db.users.create_index("uid", unique=True, name="unique uid")
+    db.users.create_index("username", unique=True, name="unique usernames")
+    db.users.create_index("username", unique=True, collation=Collation(
+            locale="en", strength=CollationStrength.PRIMARY
+        ), name="unique normalized usernames")
+    db.users.create_index("tid")
```
Comment (hiliang-cmu) [inline picoCTF-web/api/db.py]: I think some of this is duped with the users indexes below. I also added indexes for the demo.parentemail, email in consideration of the management panel user search, but that kinda goes to question if we really need such a thing for one-off manual queries like that

Comment (hiliang-cmu) [reply picoCTF-web/api/db.py]: Or rather, should be considered when this step 1: INDEX ALL THE THINGS moves to step 2: trim unnecessary indexes

**Comment by dmartin** [comment]
Updating a running server with this PR will require a manual DB fix due to f3c907d4d0b9dd06817433b2f6a9c5f80a781f03 (no settings doc with settings_id: 1 will initially exist, and a new one will be created)

**Comment by hiliang-cmu** [comment]
Was curious and checked, it takes 30ms to run through all the index creation attempts. Fairly minimal overhead.

**Pull request merged**